core/src/main/resources/static/js/modifyEnvs.js

```diff
@@ -158,7 +158,19 @@ app.controller("modifyEnvsCtrl", function($scope, $http, $location, $window) {
                     data: {'envSelected' : envId, 'envType' : envType}
                 }).success(function(output) {
                     if(output != null && output !== ""){
+
                         $scope.envDetails = output;
+
+                        $scope.envDetails.topicRegex = output.params.topicRegex[0];
+                        $scope.envDetails.topicSuffix = output.params.topicSuffix[0];
+                        $scope.envDetails.topicPrefix = output.params.topicPrefix[0];
+
+                        $scope.envDetails.defaultPartitions = output.params.defaultPartitions[0];
+                        $scope.envDetails.maxPartitions = output.params.partitionsList.length;
+                        $scope.envDetails.defaultReplicationFactor = output.params.defaultRepFactor[0];
+                        $scope.envDetails.maxReplicationFactor = output.params.replicationFactorList.length;
+                        $scope.envDetails.advancedTopicConfiguration = JSON.parse(output.params.advancedTopicConfiguration[0]);
```

Comment:
if its string list, is it for future ?

Reply:
It is a String List, I think there was a technical reason I stored it like that but I can go back and double check I might have resolved that separately.